extension/src/background/helpers/dataStorage.ts

```diff
@@ -107,9 +107,16 @@ export const migrateTokenIdList = async () => {
   const storageVersion = (await localStore.getItem(STORAGE_VERSION)) as string;
 
   if (shouldRunMigration({ storageVersion, migrationVersion: "1.0.0" })) {
-    if (Array.isArray(tokenIdsByKey)) {
+    try {
+      if (Array.isArray(tokenIdsByKey)) {
+        const newTokenList = {
+          [NETWORKS.FUTURENET]: tokenIdsByKey,
+        };
+        await localStore.setItem(TOKEN_ID_LIST, newTokenList);
+      }
+    } catch (error) {
       const newTokenList = {
-        [NETWORKS.FUTURENET]: tokenIdsByKey,
+        [NETWORKS.FUTURENET]: [],
       };
```

Comment:
What error are we expecting to catch here?

Reply:
I also thought users with the migration issues had gotten past these versions fine, whats the rationale for updating their logic?